libvips/iofuncs/semaphore.c

```diff
@@ -144,3 +144,43 @@ vips_semaphore_down( VipsSemaphore *s )
 {
 	return( vips_semaphore_downn( s, 1 ) );
 }
+
+/* Waits until sem > 0, then decrement.
+ * Returns -1 when the relative time in @timeout_ms (in microseconds)
+ * was passed.
+ */
+int
+vips__semaphore_down_timed( VipsSemaphore *s, double timeout_ms )
```

Comment:
How about just using seconds? With a double, the unit doesn't really affect the accuracy, so I think I'd expect the most human-friendly unit. If it's really microseconds, a gint64 would probably be better.

Reply:
Commit 78e251dd7504c860ace6e8e3195127107701c5c6 makes this a `gint64` instead.